src/Source/CSVFileSource.php

```diff
@@ -32,7 +32,7 @@ public function getMappedData(): array
 
         while ($current = $csv->current()) {
             // TODO make configurable?
-            $current = array_map('utf8_encode', $current);
+            $current = array_map(function ($element) {return mb_convert_encoding($element, 'UTF-8'); }, $current);
```

Comment:
Mhm, I personally have no idea if that's correct in all cases. The TODO is legit I think since it depends on the encoding type of your csv file, no?

Where's exactly the difference? I didn't have problems using the existing implementation.

CC @koertho

Reply:
utf8_encode only supports ISO-8859-1 as source encoding, mb_convert_encoding has a wider support. So I think this is a good change.

Reply:
@Defcon0 i experinced wrong  converting when the csv is having different encoding for different collumns, so the column with umlauts was already utf-8. But utf8_encode() encoded it again and convertet it back into ascii.

with the function mb_convert_encoding it should be allways the encoding we want. But idont know if it breaks on something aswell.

We can even use 
`$current = mb_convert_encoding($current, 'UTF-8');`
here, since it can even handle array of strings

Reply:
OK, then I'm fine with it. Nice finding!